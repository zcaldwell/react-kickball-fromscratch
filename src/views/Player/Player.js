import React from 'react';
import { useState, useEffect } from 'react';
import { fetchPlayerById } from '../../services/players';
import PlayerDetail from '../../components/PlayerDetail';

export default function Player(props) {
  const [player, setPlayer] = useState({ player: [] });
  console.log(player);
  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchPlayerById(props.match.params.id);
      setPlayer(data[0]);
    };
    fetchData();
  }, [props]);

  return (
    <>
      <PlayerDetail player={player} />
    </>
  );
}
