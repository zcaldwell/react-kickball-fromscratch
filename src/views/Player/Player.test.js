import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Player from './Player';

test('should render player details', async () => {
  const { container } = render(
    <MemoryRouter>
      <Player match={{ params: { id: 12 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Betty Grey');
  expect(container).toMatchSnapshot();
});
