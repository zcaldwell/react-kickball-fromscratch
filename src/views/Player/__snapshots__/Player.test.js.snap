// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render player details 1`] = `
<div>
  <div>
    <h2>
      Betty Grey
    </h2>
    <h3>
      Shortstop
    </h3>
  </div>
</div>
`;
